### PR TITLE
First version of network operator

### PR DESCRIPTION
**Pull request by siredmar**
Deploying:
- secrets for applications received from credsmanager
- nats daemonset for edge
- node-dns daemonset for edge
- credsmanager

### CODE REVIEW

File: controllers/network_controller.go [hunk 1]
```diff
@@ -43,11 +44,12 @@ var (
...
-	credsmanagerServicePort = 6000
-	credsmanagerNamespace   = "edgefarm-network"
-	timeoutSeconds          = 10
-	natsCredsSecretName     = "nats-credentials"
+	credsmanagerServiceName          = "credsmanager"
+	credsmanagerServicePort          = 6000
+	credsmanagerNamespace            = "edgefarm-network"
+	timeoutSeconds                   = 10
+	natsCredsSecretName              = "nats-credentials"
+	edgefarmNetworkAccountNameSecret = "edgefarm.network-natsAccount"
```
Comment (KairaMiu) [inline controllers/network_controller.go:52]: I would prefer some more fitting name like:
```
edgefarmNetworkUserSecret = "edgefarm.network-natsUserData"
```
But to be honest, I imagined the structure of the secret as follows:
```
$ k get secrets mixed-app.edge-worker-1 -o yaml
apiVersion: v1
data:
  userAccountName: mixed-app.edge-worker-1
  username: edge-worker-1 
  password: LxA...uD
  creds: U...KioqKioqKlxuIn0=
kind: Secret
...
```

File: pkg/additional/nats.go [hunk 2]
```diff
@@ -0,0 +1,291 @@
...
+		return err
+	}
+
+	configMap := corev1.ConfigMap{
+		TypeMeta: metav1.TypeMeta{
+			Kind:       "ConfigMap",
+			APIVersion: "v1",
+		},
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      "nats-config",
```
Comment (KairaMiu) [inline pkg/additional/nats.go:34]: Please call it `nats-init-config` to make it clear that this config is only used as initial configuration of the nats server.

**Review by KairaMiu — COMMENTED** [review]
Do not forget to use edgefarm.network from main branch (when merged)
- go.mod
- devspace.yaml

File: pkg/additional/nats.go (hunk 2 above)
Comment (siredmar) [reply pkg/additional/nats.go:34]: You're right. This can be quite confusing.

File: controllers/network_controller.go (hunk 1 above)
Comment (siredmar) [reply controllers/network_controller.go:52]: I will rename like you proposed. 
Keeping the secret in your imagined form has the following implications:
Each section in the secret will create a file called e.g. `userAccountName` with the content. So there will be four files that need to mounted, and parsed seperately.
Imho there is no need for an easy human readable form like you propoese. This secret needs to be read by applications. Keeping the secret in a json format, enables the application to easily unmarshall the secret and access the parts.
Are you okay with that?

**Comment by siredmar** [comment]
> Do not forget to use edgefarm.network from main branch (when merged)
> 
> * go.mod
> * devspace.yaml

I will provide another PR for the first release after this PR is done.

File: controllers/network_controller.go (hunk 1 above)
Comment (KairaMiu) [reply controllers/network_controller.go:52]: yes

**Review by KairaMiu — APPROVED** [review]

**Pull request merged**